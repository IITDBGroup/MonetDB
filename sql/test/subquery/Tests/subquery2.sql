
CREATE TABLE students(id INTEGER, name VARCHAR(128), major VARCHAR(128), "year" INTEGER);
CREATE TABLE exams(sid INTEGER, course VARCHAR(128), curriculum VARCHAR(128), grade INTEGER, "year" INTEGER);
INSERT INTO students VALUES (1, 'Mark', 'CS', 2017);
INSERT INTO students VALUES (2, 'Dirk', 'CS', 2017);
INSERT INTO exams VALUES (1, 'Database Systems', 'CS', 10, 2015);
INSERT INTO exams VALUES (1, 'Graphics', 'CS', 9, 2016);
INSERT INTO exams VALUES (2, 'Database Systems', 'CS', 7, 2015);
INSERT INTO exams VALUES (2, 'Graphics', 'CS', 7, 2016);

SELECT s.name, e.course, e.grade FROM students s, exams e WHERE s.id=e.sid AND e.grade=(SELECT MAX(e2.grade) FROM exams e2 WHERE s.id=e2.sid) ORDER BY name, course;
	-- Dirk, Database Systems, 7
	-- Dirk, Graphics, 7
	-- Mark, Database Systems, 10

SELECT s.name, e.course, e.grade FROM students s, exams e WHERE s.id=e.sid AND (s.major = 'CS' OR s.major = 'Games Eng') AND e.grade <= (SELECT AVG(e2.grade) - 1 FROM exams e2 WHERE s.id=e2.sid OR (e2.curriculum=s.major AND s."year">=e2."year")) ORDER BY name, course;
	-- Dirk, Database Systems, 7
	-- Dirk, Graphics, 7

SELECT name, major FROM students s WHERE EXISTS(SELECT * FROM exams e WHERE e.sid=s.id AND grade=10) OR s.name='Dirk' ORDER BY name;
	-- Dirk, CS
	-- Mark, CS

drop table students;
drop table exams;

CREATE TABLE tbl_ProductSales (ColID int, Product_Category  varchar(64), Product_Name  varchar(64), TotalSales int); 
INSERT INTO tbl_ProductSales VALUES (1,'Game','Mobo Game',200),(2,'Game','PKO Game',400),(3,'Fashion','Shirt',500),(4,'Fashion','Shorts',100);
CREATE TABLE another_T (col1 INT, col2 INT, col3 INT, col4 INT, col5 INT, col6 INT, col7 INT, col8 INT);
INSERT INTO another_T VALUES (1,2,3,4,5,6,7,8), (11,22,33,44,55,66,77,88), (111,222,333,444,555,666,777,888), (1111,2222,3333,4444,5555,6666,7777,8888);
SELECT col1 IN (SELECT ColID + col1 FROM tbl_ProductSales) FROM another_T GROUP BY col1; 
	-- False
	-- False
	-- False
	-- False

INSERT INTO tbl_ProductSales VALUES (0, 'a', 'b', 0);
SELECT col1 IN (SELECT ColID + col1 FROM tbl_ProductSales) FROM another_T GROUP BY col1; 
	-- True
	-- True
	-- True
	-- True

DROP TABLE tbl_ProductSales;
DROP TABLE another_T;
