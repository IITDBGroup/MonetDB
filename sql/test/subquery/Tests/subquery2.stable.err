stderr of test 'subquery2` in directory 'sql/test/subquery` itself:


# 18:55:45 >  
# 18:55:45 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=39342" "--set" "mapi_usock=/var/tmp/mtest-31262/.s.monetdb.39342" "--set" "monet_prompt=" "--forcemito" "--dbpath=/home/niels/scratch/monetdb/Linux-x86_64/var/MonetDB/mTests_sql_test_subquery" "--set" "embedded_c=true"
# 18:55:45 >  

# builtin opt 	gdk_dbpath = /home/niels/scratch/monetdb/Linux-x86_64/var/monetdb5/dbfarm/demo
# builtin opt 	monet_prompt = >
# builtin opt 	monet_daemon = no
# builtin opt 	mapi_port = 50000
# builtin opt 	mapi_open = false
# builtin opt 	mapi_autosense = false
# builtin opt 	sql_optimizer = default_pipe
# builtin opt 	sql_debug = 0
# cmdline opt 	gdk_nr_threads = 0
# cmdline opt 	mapi_open = true
# cmdline opt 	mapi_port = 39342
# cmdline opt 	mapi_usock = /var/tmp/mtest-31262/.s.monetdb.39342
# cmdline opt 	monet_prompt = 
# cmdline opt 	gdk_dbpath = /home/niels/scratch/monetdb/Linux-x86_64/var/MonetDB/mTests_sql_test_subquery
# cmdline opt 	embedded_c = true
#client1:!ERROR:MALException:client.quit:Server stopped

# 18:55:46 >  
# 18:55:46 >  "mclient" "-lsql" "-ftest" "-tnone" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-31262" "--port=39342"
# 18:55:46 >  


# 18:55:46 >  
# 18:55:46 >  "Done."
# 18:55:46 >  

