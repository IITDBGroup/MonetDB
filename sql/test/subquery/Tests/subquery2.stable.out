stdout of test 'subquery2` in directory 'sql/test/subquery` itself:


# 18:55:45 >  
# 18:55:45 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=39342" "--set" "mapi_usock=/var/tmp/mtest-31262/.s.monetdb.39342" "--set" "monet_prompt=" "--forcemito" "--dbpath=/home/niels/scratch/monetdb/Linux-x86_64/var/MonetDB/mTests_sql_test_subquery" "--set" "embedded_c=true"
# 18:55:45 >  

# MonetDB 5 server v11.34.0 (hg id: 179f82506958+)
# This is an unreleased version
# Serving database 'mTests_sql_test_subquery', using 8 threads
# Compiled for x86_64-unknown-linux-gnu/64bit with 128bit integers
# Found 15.385 GiB available main-memory.
# Copyright (c) 1993 - July 2008 CWI.
# Copyright (c) August 2008 - 2019 MonetDB B.V., all rights reserved
# Visit https://www.monetdb.org/ for further information
# Listening for connection requests on mapi:monetdb://da200:39342/
# Listening for UNIX domain connection requests on mapi:monetdb:///var/tmp/mtest-31262/.s.monetdb.39342
# MonetDB/GIS module loaded
# MonetDB/SQL module loaded

# 18:55:46 >  
# 18:55:46 >  "mclient" "-lsql" "-ftest" "-tnone" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-31262" "--port=39342"
# 18:55:46 >  

#CREATE TABLE students(id INTEGER, name VARCHAR(128), major VARCHAR(128), "year" INTEGER);
#CREATE TABLE exams(sid INTEGER, course VARCHAR(128), curriculum VARCHAR(128), grade INTEGER, "year" INTEGER);
#INSERT INTO students VALUES (1, 'Mark', 'CS', 2017);
[ 1	]
#INSERT INTO students VALUES (2, 'Dirk', 'CS', 2017);
[ 1	]
#INSERT INTO exams VALUES (1, 'Database Systems', 'CS', 10, 2015);
[ 1	]
#INSERT INTO exams VALUES (1, 'Graphics', 'CS', 9, 2016);
[ 1	]
#INSERT INTO exams VALUES (2, 'Database Systems', 'CS', 7, 2015);
[ 1	]
#INSERT INTO exams VALUES (2, 'Graphics', 'CS', 7, 2016);
[ 1	]
#SELECT s.name, e.course, e.grade FROM students s, exams e WHERE s.id=e.sid AND e.grade=(SELECT MAX(e2.grade) FROM exams e2 WHERE s.id=e2.sid) ORDER BY name, course;
% sys.s,	sys.e,	sys.e # table_name
% name,	course,	grade # name
% varchar,	varchar,	int # type
% 4,	16,	2 # length
[ "Dirk",	"Database Systems",	7	]
[ "Dirk",	"Graphics",	7	]
[ "Mark",	"Database Systems",	10	]
#SELECT s.name, e.course, e.grade FROM students s, exams e WHERE s.id=e.sid AND (s.major = 'CS' OR s.major = 'Games Eng') AND e.grade <= (SELECT AVG(e2.grade) - 1 FROM exams e2 WHERE s.id=e2.sid OR (e2.curriculum=s.major AND s."year">=e2."year")) ORDER BY name, course;
% sys.s,	sys.e,	sys.e # table_name
% name,	course,	grade # name
% varchar,	varchar,	int # type
% 4,	16,	1 # length
[ "Dirk",	"Database Systems",	7	]
[ "Dirk",	"Graphics",	7	]
#SELECT name, major FROM students s WHERE EXISTS(SELECT * FROM exams e WHERE e.sid=s.id AND grade=10) OR s.name='Dirk' ORDER BY name;
% .s,	.s # table_name
% name,	major # name
% varchar,	varchar # type
% 4,	2 # length
[ "Dirk",	"CS"	]
[ "Mark",	"CS"	]
#drop table students;
#drop table exams;

# 18:55:46 >  
# 18:55:46 >  "Done."
# 18:55:46 >  

